src/app/components/pages/IsaacCompetition/IsaacCompetition.tsx

```diff
@@ -31,6 +33,41 @@ export const IsaacCompetition = () => {
     setOpen(id ?? null);
   };
 
+  const user = useAppSelector(selectors.user.orNull);
```

Comment:
Consider refactoring the `renderEntryForm` function to improve readability by extracting the student and default cases into separate components. Additionally, harmonise the use of Bootstrap classes and avoid magic strings in `setOpen("5")` by using constants. This will enhance maintainability and clarity.